### ISSUE TITLE
Inability to Reproduce Paper Results

### ISSUE THREAD

**Issue by sh237** [anonrabit/DialFRED#9]
Hi, thank you for sharing the code for running the DialFRED models.

We are trying to reproduce the results from the paper for the validation unseen set, specifically an SR of 33.6 (RL Anytime) as shown in Table 1 of the paper (https://arxiv.org/pdf/2202.13330.pdf). We followed the instructions in the README.md file and executed "python train_eval.py --mode eval", which resulted in the following output:

INFO:root:78m 22s (- -20m 2s) (1363 134%) reward -0.0506, SR 0.2032, pws 0.1234

However, there seems to be a discrepancy between the score shown in the paper and the reproduced results. Could you please confirm if this is an expected result?

**Comment by jiminsun** [comment]
I also got similar results using the shared checkpoints -- SR 0.2057 and PWS 0.1240 for the unseen dev set. 
Since I used the checkpoints provided by the shell script, I expect that this should be sufficient to reproduce the numbers on paper -- is there anything that I might have missed?

**Comment by xfgao** [comment]
Sorry for the confusion caused. 

You can evaluate the finetuned questioner checkpoint by: 
`python train_eval.py --mode eval --questioner-path ./logs/pretrained/questioner_anytime_finetuned.pt`

It should produce results similar to the ones reported in the paper.

**Comment by sh237** [comment]
Thank you for your response! We tried running the code you suggested:

`python train_eval.py --mode eval --questioner-path ./logs/pretrained/questioner_anytime_finetuned.pt`

We checked the last line of logs/rl_anytime_eval_unseen1.log, which showed the following output:

INFO:root:172m 5s (- -44m 11s) (1363 134%) reward -0.0471, SR 0.2018, pws 0.1218

We were still unable to reproduce the reported results in the paper. We would greatly appreciate your continued assistance with this issue.

**Comment by xfgao** [comment]
Thanks for letting me know. 

I think this is likely due to a bug in `append_data.py` caused by a recent code refactoring. 

Can you try the following things: 
1) pull the latest commit
2) remove the old `data/lmdb_augmented_human_subgoal` folder
3) run `python append_data.py`
4) create the lmdb dataset again
```
export EXP_NAME=augmented_human
export SUBGOAL=subgoal
export EVAL_TYPE=valid_unseen

python -m alfred.data.create_lmdb with args.visual_checkpoint=$LOGS/pretrained/fasterrcnn_model.pth args.data_output=lmdb_${EXP_NAME}_${SUBGOAL} args.vocab_path=$DF_ROOT/files/$EXP_NAME.vocab > ./logs/et_${EXP_NAME}_${SUBGOAL}.log 2>&1 &
```
5) run the evaluation
```
python train_eval.py --mode eval --questioner-path ./logs/pretrained/questioner_anytime_finetuned.pt
```

As a result, I was able to get a 33.3% SR on my side.

**Comment by sh237** [comment]
We ran the code you suggested and were able to obtain the following results:

`INFO:root:66m 49s (- -18m 59s) (1363 134%) reward 0.1226, SR 0.3324, pws 0.2034`

We greatly appreciate your time and valuable assistance in helping us achieve these results. Thank you!
